go/mysql/conn.go

```diff
@@ -1210,19 +1239,108 @@ func (c *Conn) handleNextCommand(handler Handler) error {
 			log.Error("Error writing ComStmtReset OK packet to client %v: %v", c.ConnectionID, err)
 			return err
 		}
-
 	case ComStmtFetch:
-		_, _, _ = c.parseComStmtFetch(data)
+		log.Info("Received COM_STMT_FETCH")
+		if c.cs == nil {
+			log.Error("Fetching from missing result set. Client %v, returning error: %v", c.ConnectionID, data)
+			if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
+				log.Error("Error writing error packet to client: %v", err)
+				return err
+			}
+		}
+
+		c.startWriterBuffering()
+
+		stmtID, numRows, ok := c.parseComStmtFetch(data)
 		c.recycleReadPacket()
+		if !ok {
+			log.Error("Got unhandled packet from client %v, returning error: %v", c.ConnectionID, data)
+			if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
+				log.Error("Error writing error packet to client: %v", err)
+				return err
+			}
+		}
 
-		err = fmt.Errorf("COM_STMT_FETCH is not suppported")
-		if werr := c.writeErrorPacketFromError(err); werr != nil {
-			// If we can't even write the error, we're done.
-			log.Errorf("Error writing query error to %s: %v", c, werr)
-			return werr
+		if stmtID != c.cs.stmtID {
+			log.Errorf("Requested stmtID does not match stmtID of open cursor. Client %v, returning error: %v", c.ConnectionID, data)
+			if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
+				log.Error("Error writing error packet to client: %v", err)
+				return err
+			}
+		}
+
+		// If fetching, we definitely have a cursor open
+		c.StatusFlags |= uint16(ServerCursorExists)
+
+		// while we are still expecting rows
+		for !c.cs.noData && (c.cs.pending == nil || len(c.cs.pending.Rows) < int(numRows)) {
+			// drain rows from next to cs.pending
+			var newqr *sqltypes.Result
+			newqr, ok = <-c.cs.next
+
+			// channel is closed, meaning there are no more rows
+			if !ok {
+				// check if there was an error
+				if err = <- c.cs.done; err != nil {
+					return err
```

Comment:
I'm not sure if this is right...at least in the case where we haven't sent anything, it seems like we would want to convert this error into a `writeErrorPacketFromError` call, instead of returning this error as if it was an I/O error on the connection itself. I'm not sure if MySQL would first send along the rows it got and then return this error in the next Fetch? We should write some tests or something...